core/common/src/main/java/alluxio/underfs/UnderFileSystemConfiguration.java

```diff
@@ -23,14 +23,21 @@
  * {@link RuntimeException} if the key is not found in both configurations..
  */
 public final class UnderFileSystemConfiguration {
+  private final boolean mReadOnly;
+  private final boolean mShared;
```

Comment:
technically, this is mounting information regarding how Alluxio treating operations into these locations. Previously we thought this information is not necessary to propagate to UFS but keeping it in master is sufficient. However, it seems we have a usecase now. So the question is, how to represent this mounting or Alluxio filesystem related information to ufs adaptors? @madanadit creates two additional fields here in UfsConf. Maybe we should simply put `MountInfo` in here?